### PR TITLE
Messages refactor

### PR DESCRIPTION
**Pull request by jiivan**
Migrate all messages to mapped version.

### CODE REVIEW

**Review by banasiakadam60 — APPROVED** [review]

File: golem/network/p2p/p2pservice.py [hunk 1]
```diff
@@ -513,7 +513,10 @@ def find_node(self, node_key_id):
         :param node_key_id: public key of a sought node
         :return list: list of information about closest neighbours
         """
-        neighbours = self.peer_keeper.neighbours(node_key_id)
+        if node_key_id is None:
```
Comment (badb) [inline golem/network/p2p/p2pservice.py:516]: What is the purpose of this change? When will we call this method with `node_key_id` equal to `None`?

It's quite confusing for me to connect method that sends information about currently known /connected peers with Kademlia's `find_node` method.

Comment (jiivan) [reply golem/network/p2p/p2pservice.py:516]: Here `node_key_id` is `None` https://github.com/golemfactory/golem/pull/1037/files/e18d8804ff93a577c95a3a4ff606d58c396208d4#diff-9c27e4ebfe453659315cc78d4e9532e9R324

I wanted to refactor copy&paste methods that just composed `nodes_info` and sent `MessagePeers`.

**Review by Elfoniok — APPROVED** [review]

File: golem/network/p2p/peersession.py [hunk 2]
```diff
@@ -430,25 +424,11 @@ def __send_hello(self):
...
-    def __send_tasks(self, tasks):
-        self.send(MessageTasks(tasks))
-
-    def __send_resource_peers(self):
-        resource_peers = self.p2p_service.get_resource_peers()
-        self.send(MessageResourcePeers(resource_peers))
+        self.send(message.MessagePing())
+
+    def _send_peers(self, node_key_id=None):
+        nodes_info = self.p2p_service.find_node(node_key_id=node_key_id)
```
Comment (badb) [inline golem/network/p2p/peersession.py:430]:
```
  return callWithContext({"system": lp}, func, *args, **kw)
  File "C:\Python27\lib\site-packages\twisted\python\log.py", line 86, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File "C:\Python27\lib\site-packages\twisted\python\context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "C:\Python27\lib\site-packages\twisted\python\context.py", line 85, in callWithContext
    return func(*args,**kw)
--- <exception caught here> ---
  File "C:\Python27\lib\site-packages\twisted\internet\iocpreactor\reactor.py", line 119, in _callEventCallback
    evt.callback(rc, numBytes, evt)
  File "C:\Python27\lib\site-packages\twisted\internet\iocpreactor\abstract.py", line 92, in _cbRead
    if self._handleRead(rc, data, evt):
  File "C:\Python27\lib\site-packages\twisted\internet\iocpreactor\abstract.py", line 118, in _handleRead
    return self._dispatchData()
  File "C:\Python27\lib\site-packages\twisted\internet\iocpreactor\abstract.py", line 75, in _dispatchData
    self.dataReceived(self._readBuffers[full_buffers][0:remainder])
  File "C:\Python27\lib\site-packages\twisted\internet\iocpreactor\tcp.py", line 75, in dataReceived
    self.protocol.dataReceived(rbuffer)
  File "C:\Python27\lib\site-packages\twisted\internet\endpoints.py", line 119, in dataReceived
    return self._wrappedProtocol.dataReceived(data)
  File "C:\Sources\golem\golem\network\transport\tcpnetwork.py", line 476, in dataReceived
    self._interpret(data)
  File "C:\Sources\golem\golem\network\transport\tcpnetwork.py", line 505, in _interpret
    self.session.interpret(m)
  File "C:\Sources\golem\golem\network\p2p\peersession.py", line 87, in interpret
    BasicSafeSession.interpret(self, msg)
  File "C:\Sources\golem\golem\network\transport\session.py", line 96, in interpret
    action(msg)
  File "C:\Sources\golem\golem\network\p2p\peersession.py", line 325, in _react_to_get_peers
    self._send_peers()
  File "C:\Sources\golem\golem\network\p2p\peersession.py", line 430, in _send_peers
    nodes_info = self.p2p_service.find_node(node_key_id=node_key_id)
  File "C:\Sources\golem\golem\network\p2p\p2pservice.py", line 521, in find_node
    for peer in neighbours:
exceptions.TypeError: 'builtin_function_or_method' object is not iterable
```

File: tests/golem/network/transport/test_tcpnetwork.py [hunk 3]
```diff
@@ -191,7 +191,7 @@ def test_dataReceived(self):
             self.assertIsNone(protocol.dataReceived(data))
         protocol.db.clear_buffer()
 
-        m = MessageDisconnect()
+        m = message.MessageDisconnect()
```
Comment (badb) [inline tests/golem/network/transport/test_tcpnetwork.py:194]: This test still fail on my machine: 
```
WARNING  [golem.network.transport.tcpnetwork ] No session argument in connection state
INFO     [golem.network.transport.message    ] Unrecognized message type: 2
ERROR    [golem.network.transport.message    ] Failed to deserialize message ��@�A�D&��t���qDISCONNECT_REASON 
INFO     [golem.network.transport.tcpnetwork ] Deserialization of messages from <MagicMock name='mock.address' id='99130576'>:<MagicMock name='mock.port' id='99181360'> failed, maybe it's still too short?

tests\golem\network\transport\test_tcpnetwork.py:182 (TestBasicProtocol.test_dataReceived)
self = <test_tcpnetwork.TestBasicProtocol testMethod=test_dataReceived>

    def test_dataReceived(self):
        data = "abc"
        protocol = BasicProtocol()
        self.assertIsNone(protocol.dataReceived(data))
        protocol.opened = True
        self.assertIsNone(protocol.dataReceived(data))
        protocol.session = MagicMock()
        with self.assertLogs(logger):
            self.assertIsNone(protocol.dataReceived(data))
        protocol.db.clear_buffer()
    
        m = message.MessageDisconnect()
        data = m.serialize()
        packed_data = struct.pack("!L", len(data)) + data
        protocol.dataReceived(packed_data)
>       self.assertEqual(protocol.session.interpret.call_args[0][0].TYPE, m.TYPE)
E       TypeError: 'NoneType' object has no attribute '__getitem__'

test_tcpnetwork.py:198: TypeError
```

**Review by badb — CHANGES_REQUESTED** [review]

**Review by badb — APPROVED** [review]

**Pull request merged**